src/pages/payment/payment.shared.tsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from 'react';
+import { useMatch } from '@tanstack/react-location';
+import { useAccount } from 'wagmi';
+import Dapp from 'src/dapp';
+import store from 'src/store/store';
+import { hideSpinner, showSpinner } from 'src/store/reducers/spinner.reducer';
+import { endpoint } from 'src/core/endpoints';
+import { confirmPayment, getCreditCardInfo } from './payment.service';
+import { dialog } from 'src/core/dialog/dialog';
+import { getMonthName } from 'src/core/time';
+import { Resolver } from './payment.types';
+
+export const usePaymentShared = () => {
+  const { web3 } = Dapp.useWeb3();
+  const { address: account, isConnected } = useAccount();
+  const { offer, cardInfo } = useMatch().ownData as Resolver;
+  const [process, setProcess] = useState(false);
+  const [selectedCard, setSelectedCard] = useState(cardInfo?.items[0]?.id);
+  const [cards, setCards] = useState(cardInfo);
+  const offerId = offer.id;
+  const {
+    created_at,
+    recipient: {
+      meta: { wallet_address: contributor },
+    },
+    assignment_total,
+    project_id,
+    project: { payment_type },
+    payment_mode,
+  } = offer;
+  const commision = assignment_total * 0.03;
+  const total_price = commision + assignment_total;
+  const start_date = getMonthName(created_at) + ' ' + new Date(created_at).getDate();
+  const isPaidCrypto = payment_type === 'PAID' && payment_mode === 'CRYPTO';
+  const isDisabledProceedPayment = process || (isPaidCrypto ? !isConnected || !account : !selectedCard);
+
+  function onSelectCard(id: string) {
+    setSelectedCard(id);
+  }
+
+  async function onRemoveCard(id: string) {
+    setSelectedCard('');
+    endpoint.post.payments['{card_id}/remove'](id);
+    try {
+      const result = await getCreditCardInfo();
+      setCards(result);
+    } catch (err: any) {
```

Comment:
Catching the error should be handled in our http interceptor

Reply:
whenever it is handled, I can remove these try/catch because for now, I need them to show the user if something happened. Also as I talked with you there is no problem for now if we use `dialog.alert`